include/roboptim/trajectory/visualization/matplotlib.hh

```diff
@@ -0,0 +1,152 @@
+
+          ss << data_name << " = np.array ([";
+
+          for (; n >= 0.; n--, t += step)
+          {
+            Function::vector_t res = spline (t);
+
+            ss << (boost::format ("(%2.8f") % normalize (t)).str ();
+
+            for (Function::size_type i = 0; i < spline.outputSize(); ++i)
+            {
+              ss << (boost::format (", %2.8f") % normalize (res[i])).str();
+            }
+            ss << "), ";
+          }
+          ss << "])" << std::endl;
+
+          ss << (boost::format("plt.plot(%1%[:,1], min(%1%[:,2]), lw=2, color='c')")
+                 % data_name).str() << std::endl;
+
+          ss << (boost::format("plt.plot(%1%[:,1], max(%1%[:,2]), lw=2, color='g')")
+                 % data_name).str() << std::endl;
+
+          if (spline.outputSize() != 2)
+          {
+            ;
+          }
+          else
+          {
+            ss << (boost::format("plt.plot(%1%[:,1], %1%[:,2], label=\"%2%\", color='b')")
+                   % data_name
+                   % (boost::format("%1% on interval [%2%, %3%]")
+                      % spline.getName()
+                      % bound_min
+                      % bound_max).str()).str() << std::endl;
+          }
+          ss << (boost::format("plt.axvline(%1%[-1,1], ls='--', color='gray')")
+                % data_name).str() << std::endl;
+
+          return ss.str();
+        }
+
+        Command plot_spline(const CubicBSpline& spline,
+                            CubicBSpline::value_type step)
+        {
+          using boost::format;
+          using namespace detail;
+          using namespace ::roboptim::visualization;
+
+          std::stringstream ss;
+          long dimension = spline.outputSize();
+          double start = spline.getLowerBound(spline.timeRange ());
+          double end = spline.getUpperBound(spline.timeRange ());
+          CubicBSpline::size_type nbp = spline.getNumberControlPoints();
+          CubicBSpline::size_type intervals = nbp - 3;
+
+          if (dimension != 2)
+          {
+            std::cerr << "This tool is only designed to print 2D splines"
+              << std::endl;
+          }
+
+          double inc = (end-start)/static_cast<double>(intervals);
+
+          if (inc <= step)
+          {
+            throw std::string("inadapted step");
+          }
+          ss << "def min(x):" << std::endl;
+          ss << "  return np.min(x)*np.ones(np.shape(x))" << std::endl;
+
+          ss << "def max(x):" << std::endl;
+          ss << "  return np.max(x)*np.ones(np.shape(x))" << std::endl;
+
+          ss << "CP = np.array([";
+
+          for (int i = 0; i < nbp; ++i)
+            ss << (boost::format("(%1%, %2%),")
+                 % spline.parameters()[2*i]
+                 % spline.parameters()[2*i+1]).str();
+
+          ss << "])" << std::endl;
+
+          for (int i = 0; i < intervals; ++i, start += inc)
+          {
+            ss << interval_plot(spline, step,
```

Comment:
The spline plot may be done once rather than in a piecewise manner, and `interval_plot` would simply display the limit between different segments.